kedro/extras/datasets/tensorflow/tensorflow_model_dataset.py

```diff
@@ -118,15 +118,17 @@ def _load(self) -> tf.keras.Model:
                 self._fs.get(load_path, path, recursive=True)
 
             # Pass the local temporary directory/file path to keras.load_model
-            return tf.keras.models.load_model(path, **self._load_args)
+            device = self._load_args.pop("tf_device", "gpu")
+            if device == "cpu":
+                with tf.device("/CPU:0"):
```

Comment:
I personally think it would be nicer if this more broadly supported `tf.device`, rather than adding a special case for CPU. Maybe, if `device_name` is passed, It can use the `tf.device` context manager with specified `device_name` (that way it's not just restricted to `"/CPU:0"`); else, don't.

Reply:
Makes sense. I have updated the code to use an explicit device name from `load_args`